### PR TITLE
[zh-cn]: Update translation of String.anchor()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#13975

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md [hunk 1]
```diff
@@ -3,51 +3,43 @@ title: String.prototype.anchor()
 slug: Web/JavaScript/Reference/Global_Objects/String/anchor
 ---
 
-{{JSRef("Global_Objects", "String")}}
+{{JSRef}} {{deprecated_header}}
 
-## 概述
+{{jsxref("String")}} 的 **`anchor()`** 方法创建一个字符串，将该字符串嵌入到一个带有名称的 {{HTMLElement("a")}} 元素中（`<a name="...">str</a>`）。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md:8]: 这段的翻译感觉怪怪的，改进一下

File: files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md [hunk 2]
```diff
@@ -3,51 +3,43 @@ title: String.prototype.anchor()
...
 
-> **警告：** 该特性已经从 Web 标准中删除，虽然一些浏览器目前仍然支持它，但也许会在未来的某个时间停止支持，请尽量不要使用该特性。
-
-**`anchor()`** 方法创建一个 {{HTMLElement("a")}} HTML 锚元素，被用作超文本靶标（hypertext target）。
+> **备注：** 所有的 [HTML 包装方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#html_包装器方法)都已被弃用，并且仅为了兼容性而标准化。请使用 [DOM API](/zh-CN/docs/Web/API/Document_Object_Model)（比如 [`document.createElement()`](/zh-CN/docs/Web/API/Document/createElement) 作为代替。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md:10]:
```suggestion
> **备注：** 所有的 [HTML 包装方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#html_包装器方法)都已被弃用，并且仅为了兼容性而标准化。请使用 [DOM API](/zh-CN/docs/Web/API/Document_Object_Model)（比如 [`document.createElement()`](/zh-CN/docs/Web/API/Document/createElement)）代替。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md [hunk 3]
```diff
@@ -3,51 +3,43 @@ title: String.prototype.anchor()
...
-str.anchor(name)
+```js-nolint
+anchor(name)
 ```
 
 ### 参数
 
 - `name`
-  - : 一个字符串，表示被创建的标签的 `name` 属性。
+  - : 一个字符串，表示要生成的 `<a name="...">` 标签的 `name` 属性。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md:23]:
```suggestion
  - : 一个字符串，表示要生成的 `<a name="...">` 开始标签的 `name` 属性。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md [hunk 4]
```diff
@@ -3,51 +3,43 @@ title: String.prototype.anchor()
...
-包含 {{HTMLElement("a")}} HTML 元素的一个字符串。
-
-## 描述
-
-使用 `anchor` 方法能够以编程方式在一个文档中创建和展现一个锚链接。
-
-语法上来讲，字符串表示你想让用户看到的文本。`name` 字符串参数表示 {{HTMLElement("a")}} 元素的 `name` 属性。
-
-使用 anchor 方法创建的锚点（anchors）将会成为 {{domxref("document.anchors")}} 数组的元素。
+一个以 `<a name="name">` 开始标记开头的字符串（`name` 中的双引号被替换为 `&quot;`），然后是文本 `str`，最后是一个 `</a>` 结束标记。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md:27]:
```suggestion
一个以 `<a name="name">` 开始标签开头的字符串（`name` 中的双引号被替换为 `&quot;`），然后是文本 `str`，最后是一个 `</a>` 结束标签。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md (hunk 1 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/anchor/index.md:8]: The anchor() method of [String](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String) values creates a string that embeds this string in an [<a>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) element with a name (<a name="...">str</a>).
```suggestion
{{jsxref("String")}} 的 **`anchor()`** 方法创建一个带有名称的 {{HTMLElement("a")}} 元素字符串，其中嵌入了调用的字符串（`<a name="...">str</a>`）。
```

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t

**Pull request merged**